### PR TITLE
Real-time metrics

### PR DESCRIPTION
**Pull request by Hornwitser**
Implements basic real time collection of CPU, memory and disk usage on the systems that the controller and hosts runs on. This is not intended to replace a Prometheus, and Grafana setup, but rather provide the basic metrics to give an idea of what resources are available and being used at a glance while using the web interface. The stats collected are also exposed to Prometheus which means that for basic insights it's no longer necessary to set up a Node Exporter or similar tool to collect these.

In short this adds pretty indicator bars to the controller and hosts pages:

image

### CODE REVIEW

File: packages/controller/src/Controller.ts [hunk 1]
```diff
@@ -452,6 +507,25 @@ export default class Controller {
...
+			json = JSON.parse(await fs.readFile(filePath, { encoding: "utf8" }));
+
+		} catch (err: any) {
+			if (err.code !== "ENOENT") {
+				throw err;
+			}
+			return new Map();
+		}
+
+		return new Map(json.map(m => lib.SystemMetrics.fromJSON(m)).map(m => [m.id, m]));
```
Comment (Cooldude2606) [inline packages/controller/src/Controller.ts:522]: I find it odd that we are covering the `ENOENT` case but not the success case. Although I surpose this comes under integration tests which arent a priority.

File: packages/controller/src/routes.ts [hunk 2]
```diff
@@ -39,7 +39,7 @@ function mergeSamples(destinationResult: lib.CollectorResultSerialized, sourceRe
 	}
 
 	for (let entry of receivedSamples) {
-		sourceResult.samples.push(entry);
+		destinationResult.samples.push(entry);
```
Comment (Cooldude2606) [inline packages/controller/src/routes.ts:42]: This appears to be a bug fix, a regression test would be beneficial here.

File: packages/lib/src/data/messages_controller.ts [hunk 3]
```diff
@@ -204,6 +204,108 @@ export class LogMessageEvent {
 	}
 }
 
+export class SystemMetrics {
```
Comment (Cooldude2606) [inline packages/lib/src/data/messages_controller.ts:207]: None of the getter are tested, this is a simple one to add as you only need to call `fromJson` and then check against known good values.

**Review by Cooldude2606 — APPROVED** [review]
Only comments are about test cases so nothing needs to be added in this PR.

File: packages/controller/src/routes.ts (hunk 2 above)
Comment (Hornwitser) [reply packages/controller/src/routes.ts:42]: The commit changing the line describes it in detail. I don't think a regression test is worth the effort here because this line of code will likely never be hit.
